scripts/validationAndPreparation.sh

```diff
@@ -1,45 +1,67 @@
+#!/bin/bash
 
-FILES=`lftp -e "set ssl:key-file ~/.ssh/id_rsa; cls; bye" sftp://josojo:@trusted-setup.staging.gnosisdev.com`
-if [[ -z "${NEWESTDATE}" ]]; then
-  NEWESTDATE=0
+FILES=`lftp sftp://validationworker:@trusted-setup.staging.gnosisdev.com -e 'set sftp:connect-program "ssh -a -x -i /root/.ssh/id_rsa_worker";cls;bye'`
+
+#no contribution in folder challenges expected
+FILES=${FILES//$challenges/}
+DATEOFNEWESTCONTRIBUTION=`cat /app/config/lastestContributionDate.txt`
+if [[ -z "${DATEOFNEWESTCONTRIBUTION}" ]]; then
+  DATEOFNEWESTCONTRIBUTION=0
 fi
+
+TRUSTEDSETUPTURN=`cat /app/config/lastestContributionTurn.txt`
 if [[ -z "${TRUSTEDSETUPTURN}" ]]; then
   TRUSTEDSETUPTURN=1
 fi
+
 unset NEWESTFILE
+
+#find the newest file in the sftp server
 for f in $FILES
 do
 	echo "Processing $f"
-	DATE=` lftp -e "set ssl:key-file ~/.ssh/id_rsa; cls -l --time-style=%FT%T $f/* --sort=date | head -1; bye" sftp://josojo:@trusted-setup.staging.gnosisdev.com | awk '{print $6}' | sed 's/[^0-9]*//g'`
+	DATE=`lftp sftp://validationworker:@trusted-setup.staging.gnosisdev.com -e 'set sftp:connect-program "ssh -a -x -i /root/.ssh/id_rsa_worker"; cls -l --time-style=%FT%T '$f'/* --sort=date | head -1; bye' | awk '{print $6}' | sed 's/[^0-9]*//g'`
```

Comment:
I see this string used a couple of times 

```
'set sftp:connect-program "ssh -a -x -i /root/.ssh/id_rsa_worker"; cls -l --time-style=%FT%T '$f'/* --sort=date | head -1; bye' | awk '{print $6}'
```

Could this be given a name and referred to by name? Also, if it were given a name what would it be?

Reply:
Yeah definitively, but these aliases are so hard to implement if the commands using all the different ` " '  . This is driving me insane :dagger: